# Import PANDAS Library

Notice that we are using pd as an abbreviation so each time we want to use the PANDAS library, we can simply type pd instead of pandas.

In [1]:
import pandas as pd
pd.options.display.float_format = '{:,.2f}'.format

# Import Dataset (either full dataset or simplified)

You can use either set of datafiles (full or simplified). In the past some students' computers could not handle the full dataset. If you worry that your computer will not handle the full dataset (or if you find your code is running very slow), use the simplified version. 

The simplified version was created by taking a sample of the full data; hence the simplified files have "Sample" added to the end of the file names. Some files were not large to begin with and didn't need to be reduced. Those files do not have Sample added to the end of the file name.

## Adjust the file address below to match your computer.

You will want to replace the begining of the address below (i.e., everything before OneDrive - Oregon State University/)

In [2]:
begInv = pd.read_csv("C:/Users/Tyler Du/Downloads/SimplifiedData/SimplifiedData/HW4 Sampled Data/BegInvFINAL12312016Sample.csv")

In [3]:
begInv.head(2)

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_60      1  HARDERSFIELD     60       Canadian Club 1858 VAP   

    Size  onHand  Price   startDate  
0  750mL       8  12.99  2016-01-01  
1  750mL       7  10.99  2016-01-01

In [4]:
begInv = begInv.rename(columns = {'onHand':'01_01_2016_Qty'})
begInv.head(2)

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_60      1  HARDERSFIELD     60       Canadian Club 1858 VAP   

    Size  01_01_2016_Qty  Price   startDate  
0  750mL               8  12.99  2016-01-01  
1  750mL               7  10.99  2016-01-01

In [5]:
begInv['InvCost01_01_2016'] = begInv['01_01_2016_Qty'] * begInv['Price']
begInv.head(2)

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_60      1  HARDERSFIELD     60       Canadian Club 1858 VAP   

    Size  01_01_2016_Qty  Price   startDate  InvCost01_01_2016  
0  750mL               8  12.99  2016-01-01             103.92  
1  750mL               7  10.99  2016-01-01              76.93

In [6]:
begInv_Store = begInv[['Store','01_01_2016_Qty','InvCost01_01_2016']].groupby('Store').sum()
begInv_Store.head(2)

01_01_2016_Qty  InvCost01_01_2016
Store                                   
1               49917         838,079.59
2               52925         840,452.20

In [7]:
begInv_Store.sort_values(by='InvCost01_01_2016',ascending=False).head(10)

01_01_2016_Qty  InvCost01_01_2016
Store                                   
34             153852       3,291,170.24
73             162551       3,142,497.36
67             158996       3,079,578.63
66             149314       2,973,033.90
76             140208       2,952,418.44
69             144255       2,946,726.65
38             114368       2,232,698.77
55             119641       2,001,263.66
50              94720       1,649,808.22
79              95330       1,503,149.48

In [8]:
#alternative method for sorting below

In [9]:
begInv_Store.sort_values(by='InvCost01_01_2016').tail(10).sort_values(by='InvCost01_01_2016', ascending = False)

01_01_2016_Qty  InvCost01_01_2016
Store                                   
34             153852       3,291,170.24
73             162551       3,142,497.36
67             158996       3,079,578.63
66             149314       2,973,033.90
76             140208       2,952,418.44
69             144255       2,946,726.65
38             114368       2,232,698.77
55             119641       2,001,263.66
50              94720       1,649,808.22
79              95330       1,503,149.48

# Deliverable #1b

In [10]:
begInv_Brand = begInv[['Brand','01_01_2016_Qty','InvCost01_01_2016']].groupby('Brand').sum()
begInv_Brand.head(2)

01_01_2016_Qty  InvCost01_01_2016
Brand                                   
58                281           3,650.19
60                288           3,165.12

In [11]:
begInv_Brand.sort_values(by='InvCost01_01_2016',ascending=False).head(10)

01_01_2016_Qty  InvCost01_01_2016
Brand                                   
3545            14499         463,823.01
1233            12016         432,455.84
8068            15341         383,371.59
4261            15499         340,823.01
3858            13649         327,439.51
2753             4625         286,703.75
8082             9287         278,517.13
8680             7066         275,503.34
2589             6766         270,572.34
3876            14829         266,773.71

# Deliverable 2a

In [12]:
endInv = pd.read_csv("C:/Users/Tyler Du/Downloads/SimplifiedData/SimplifiedData/HW4 Sampled Data/EndInvFINAL12312016.csv")
endInv.head(2)

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_62      1  HARDERSFIELD     62     Herradura Silver Tequila   

    Size  onHand  Price     endDate  
0  750mL      11  12.99  2016-12-31  
1  750mL       7  36.99  2016-12-31

In [13]:
endInv = endInv.rename(columns = {'onHand':'12_31_2016_Qty'})
endInv.head(2)

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_62      1  HARDERSFIELD     62     Herradura Silver Tequila   

    Size  12_31_2016_Qty  Price     endDate  
0  750mL              11  12.99  2016-12-31  
1  750mL               7  36.99  2016-12-31

In [14]:
endInv['InvCost12_31_2016'] = endInv['12_31_2016_Qty'] * endInv['Price']
endInv.head(2)

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_62      1  HARDERSFIELD     62     Herradura Silver Tequila   

    Size  12_31_2016_Qty  Price     endDate  InvCost12_31_2016  
0  750mL              11  12.99  2016-12-31             142.89  
1  750mL               7  36.99  2016-12-31             258.93

In [15]:
endInv_Store = endInv[['Store','12_31_2016_Qty','InvCost12_31_2016']].groupby('Store').sum()
endInv_Store.head(2)

12_31_2016_Qty  InvCost12_31_2016
Store                                   
1               79827       1,206,845.93
2               56671         850,884.06

In [16]:
endInv_Store.sort_values(by='InvCost12_31_2016',ascending=False).head(10)

12_31_2016_Qty  InvCost12_31_2016
Store                                   
50             260717       4,887,260.68
73             164589       3,254,662.81
67             163765       3,076,114.82
34             145829       3,074,616.75
76             143866       2,975,945.18
69             150848       2,968,678.82
66             144579       2,860,504.99
74             166015       2,803,645.13
38             129397       2,463,906.85
55             125584       2,234,836.35

# Deliverable 2b

In [17]:
endInv_Brand = endInv[['Brand','12_31_2016_Qty','InvCost12_31_2016']].groupby('Brand').sum()
endInv_Brand.head(2)

12_31_2016_Qty  InvCost12_31_2016
Brand                                   
58                385           5,001.15
60                146           1,604.54

In [18]:
endInv_Brand.sort_values(by='InvCost12_31_2016',ascending=False).head(10)

12_31_2016_Qty  InvCost12_31_2016
Brand                                   
1233            15047         526,494.53
3545            16770         502,932.30
2753             7849         470,861.51
8068            15608         366,631.92
3405            12268         355,649.32
4261            16769         351,981.31
2757            11603         336,370.97
2589             7922         300,956.78
1376            13180         276,648.20
2585            10487         272,557.13

# Deliverable 3

In [19]:
begInv = begInv[['InventoryId','01_01_2016_Qty','InvCost01_01_2016']]
begInv.head(2)

InventoryId  01_01_2016_Qty  InvCost01_01_2016
0  1_HARDERSFIELD_58               8             103.92
1  1_HARDERSFIELD_60               7              76.93

In [20]:
begInv = begInv.rename(columns = {'01_01_2016_Qty':'BegInv_Qty'})
begInv = begInv.rename(columns = {'InvCost01_01_2016':'BegInv_Cost'})
begInv.head(2)


InventoryId  BegInv_Qty  BegInv_Cost
0  1_HARDERSFIELD_58           8       103.92
1  1_HARDERSFIELD_60           7        76.93

In [21]:
endInv = endInv[['InventoryId','12_31_2016_Qty','InvCost12_31_2016']]
endInv.head(2)

InventoryId  12_31_2016_Qty  InvCost12_31_2016
0  1_HARDERSFIELD_58              11             142.89
1  1_HARDERSFIELD_62               7             258.93

In [22]:
endInv = endInv.rename(columns = {'12_31_2016_Qty':'EndInv_Qty'})
endInv = endInv.rename(columns = {'InvCost12_31_2016':'EndInv_Cost'})
endInv.head(2)


InventoryId  EndInv_Qty  EndInv_Cost
0  1_HARDERSFIELD_58          11       142.89
1  1_HARDERSFIELD_62           7       258.93

In [23]:
Inventory = pd.merge(left = begInv, right = endInv, how = 'outer', on = ['InventoryId'])
Inventory.head()

InventoryId  BegInv_Qty  BegInv_Cost  EndInv_Qty  EndInv_Cost
0  1_HARDERSFIELD_58        8.00       103.92       11.00       142.89
1  1_HARDERSFIELD_60        7.00        76.93         NaN          NaN
2  1_HARDERSFIELD_62        6.00       221.94        7.00       258.93
3  1_HARDERSFIELD_63        3.00       116.97        7.00       272.93
4  1_HARDERSFIELD_72        6.00       209.94        4.00       139.96

In [24]:
Inventory = Inventory.fillna(0)
Inventory.head()

InventoryId  BegInv_Qty  BegInv_Cost  EndInv_Qty  EndInv_Cost
0  1_HARDERSFIELD_58        8.00       103.92       11.00       142.89
1  1_HARDERSFIELD_60        7.00        76.93        0.00         0.00
2  1_HARDERSFIELD_62        6.00       221.94        7.00       258.93
3  1_HARDERSFIELD_63        3.00       116.97        7.00       272.93
4  1_HARDERSFIELD_72        6.00       209.94        4.00       139.96

# Deliverable 4a

In [25]:
Invoices = pd.read_csv("C:/Users/Tyler Du/Downloads/SimplifiedData/SimplifiedData/HW4 Sampled Data/InvoicePurchases12312016.csv")
Invoices.head()

VendorNumber                   VendorName InvoiceDate  PONumber  \
0           105  ALTAMAR BRANDS LLC           2016-01-04      8124   
1          4466  AMERICAN VINTAGE BEVERAGE    2016-01-07      8137   
2           388  ATLANTIC IMPORTING COMPANY   2016-01-09      8169   
3           480  BACARDI USA INC              2016-01-12      8106   
4           516  BANFI PRODUCTS CORP          2016-01-07      8170   

       PODate     PayDate  Quantity    Dollars  Freight Approval  
0  2015-12-21  2016-02-16         6     214.26     3.47      NaN  
1  2015-12-22  2016-02-21        15     140.55     8.57      NaN  
2  2015-12-24  2016-02-16         5     106.60     4.61      NaN  
3  2015-12-20  2016-02-05     10100 137,483.78 2,935.20      NaN  
4  2015-12-24  2016-02-12      1935  15,527.25   429.20      NaN

In [26]:
Deliverable_4a = Invoices[["VendorNumber","Dollars"]].groupby("VendorNumber").sum()
Deliverable_4a.sort_values(by='Dollars', ascending=False)

Dollars
VendorNumber              
3960         50,959,796.85
4425         27,861,690.02
12546        24,203,151.05
17035        24,124,091.56
480          17,624,378.72
...                    ...
90033               128.64
54                  105.07
90026                77.18
1439                 54.64
201359               17.00

[126 rows x 1 columns]

# Deliverable 4b

In [27]:
Invoices.head()

VendorNumber                   VendorName InvoiceDate  PONumber  \
0           105  ALTAMAR BRANDS LLC           2016-01-04      8124   
1          4466  AMERICAN VINTAGE BEVERAGE    2016-01-07      8137   
2           388  ATLANTIC IMPORTING COMPANY   2016-01-09      8169   
3           480  BACARDI USA INC              2016-01-12      8106   
4           516  BANFI PRODUCTS CORP          2016-01-07      8170   

       PODate     PayDate  Quantity    Dollars  Freight Approval  
0  2015-12-21  2016-02-16         6     214.26     3.47      NaN  
1  2015-12-22  2016-02-21        15     140.55     8.57      NaN  
2  2015-12-24  2016-02-16         5     106.60     4.61      NaN  
3  2015-12-20  2016-02-05     10100 137,483.78 2,935.20      NaN  
4  2015-12-24  2016-02-12      1935  15,527.25   429.20      NaN

In [28]:
Deliverable_4b = Invoices[["VendorNumber","Freight"]].groupby("VendorNumber").sum()
Deliverable_4b.sort_values(by='Freight', ascending=False)

Freight
VendorNumber           
3960         257,032.07
4425         144,929.24
12546        123,880.97
17035        123,780.22
480           89,286.27
...                 ...
90033              0.61
54                 0.48
90026              0.36
1439               0.27
201359             0.09

[126 rows x 1 columns]

# Deliverable 4c

In [29]:
Deliverable_4c = pd.merge(left = Deliverable_4a, right = Deliverable_4b, how = 'inner', on = ['VendorNumber'])
Deliverable_4c.head()

Dollars  Freight
VendorNumber                   
2             5,630.88    27.08
54              105.07     0.48
60           76,770.25   367.52
105          11,706.20    62.39
200           1,205.16     6.19

In [30]:
Deliverable4c = Deliverable_4c[Deliverable_4c["Dollars"] > 250000]
Deliverable4c.head()

Dollars   Freight
VendorNumber                        
480          17,624,378.72 89,286.27
516           1,628,866.68  8,510.41
653           1,529,682.04  8,014.98
660           3,537,977.55 17,932.33
1128         13,529,433.08 68,601.68

In [31]:
Deliverable4c["Freight_per_$"] = (Deliverable4c["Freight"] / Deliverable4c["Dollars"] * 100)
Deliverable4c.head()

C:\Users\Tyler Du\AppData\Local\Temp\ipykernel_35464\3835785407.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  Deliverable4c["Freight_per_$"] = (Deliverable4c["Freight"] / Deliverable4c["Dollars"] * 100)


Dollars   Freight  Freight_per_$
VendorNumber                                       
480          17,624,378.72 89,286.27           0.51
516           1,628,866.68  8,510.41           0.52
653           1,529,682.04  8,014.98           0.52
660           3,537,977.55 17,932.33           0.51
1128         13,529,433.08 68,601.68           0.51

In [32]:
Deliverable4c.sort_values(by='Freight_per_$', ascending=False).head(10)

Dollars    Freight  Freight_per_$
VendorNumber                                        
9625            361,249.21   1,933.19           0.54
1590          1,365,472.83   7,259.75           0.53
9744            759,449.24   3,999.93           0.53
653           1,529,682.04   8,014.98           0.52
17031           300,403.20   1,573.31           0.52
516           1,628,866.68   8,510.41           0.52
4425         27,861,690.02 144,929.24           0.52
8673          3,086,650.70  15,919.70           0.52
9815          5,258,636.79  27,100.41           0.52
9165         13,210,613.93  68,054.70           0.52

In [33]:
Deliverable4c.sort_values(by='Freight_per_$', ascending=False).tail()

Dollars   Freight  Freight_per_$
VendorNumber                                      
8352         2,051,436.01 10,261.60           0.50
3924         2,816,661.94 14,069.87           0.50
3089         1,715,908.88  8,549.55           0.50
9819         2,978,686.40 14,836.57           0.50
6359           387,622.69  1,922.00           0.50

# Deliverable 4d

In [34]:
Invoices.head()

VendorNumber                   VendorName InvoiceDate  PONumber  \
0           105  ALTAMAR BRANDS LLC           2016-01-04      8124   
1          4466  AMERICAN VINTAGE BEVERAGE    2016-01-07      8137   
2           388  ATLANTIC IMPORTING COMPANY   2016-01-09      8169   
3           480  BACARDI USA INC              2016-01-12      8106   
4           516  BANFI PRODUCTS CORP          2016-01-07      8170   

       PODate     PayDate  Quantity    Dollars  Freight Approval  
0  2015-12-21  2016-02-16         6     214.26     3.47      NaN  
1  2015-12-22  2016-02-21        15     140.55     8.57      NaN  
2  2015-12-24  2016-02-16         5     106.60     4.61      NaN  
3  2015-12-20  2016-02-05     10100 137,483.78 2,935.20      NaN  
4  2015-12-24  2016-02-12      1935  15,527.25   429.20      NaN

In [35]:
Deliverable4d1 = Invoices[Invoices["Quantity"] <= 1000]
Deliverable4d1.head()

VendorNumber                   VendorName InvoiceDate  PONumber  \
0           105  ALTAMAR BRANDS LLC           2016-01-04      8124   
1          4466  AMERICAN VINTAGE BEVERAGE    2016-01-07      8137   
2           388  ATLANTIC IMPORTING COMPANY   2016-01-09      8169   
5          2396  BLACK PRINCE DISTILLERY INC  2016-01-08      8191   
7          1189  BULLY BOY DISTILLERS         2016-01-09      8171   

       PODate     PayDate  Quantity  Dollars  Freight Approval  
0  2015-12-21  2016-02-16         6   214.26     3.47      NaN  
1  2015-12-22  2016-02-21        15   140.55     8.57      NaN  
2  2015-12-24  2016-02-16         5   106.60     4.61      NaN  
5  2015-12-25  2016-02-06        23   234.83     2.30      NaN  
7  2015-12-24  2016-02-04         6   132.30     5.29      NaN

In [36]:
Deliverable4d1 = Deliverable4d1[Deliverable4d1["Freight"] > 100]
Deliverable4d1.head()


VendorNumber                   VendorName InvoiceDate  PONumber  \
11          1485  CASTLE BRANDS CORP.          2016-01-08      8152   
16          2242  DELICATO VINEYARDS INC       2016-01-06      8139   
19          2555  DISARONNO INTERNATIONAL LLC  2016-01-11      8192   
23          2561  EDRINGTON AMERICAS           2016-01-08      8175   
28          3924  HEAVEN HILL DISTILLERIES     2016-01-08      8155   

        PODate     PayDate  Quantity  Dollars  Freight Approval  
11  2015-12-23  2016-02-19       320 5,420.41   179.26      NaN  
16  2015-12-22  2016-02-10       808 6,646.46   127.05      NaN  
19  2015-12-25  2016-02-17       385 3,506.41   146.86      NaN  
23  2015-12-24  2016-02-10       136 5,645.24   218.18      NaN  
28  2015-12-23  2016-02-15       818 7,079.02   200.02      NaN

In [37]:
Deliverable4d = Deliverable4d1[["VendorNumber", "VendorName", "Freight"]].groupby("VendorNumber").sum()
Deliverable4d.sort_values(by='Freight', ascending=False).head()

VendorName  Freight
VendorNumber                                                            
2561          EDRINGTON AMERICAS         EDRINGTON AMERICAS ... 2,944.67
8664                                STOLI GROUP,(USA) LLC         349.02
7239                                REMY COINTREAU USA INC        348.07
653                                     STATE WINE & SPIRITS      303.70
6785                                PALM BAY INTERNATIONAL INC    242.73

# Deliverable 4e

In [38]:
Deliverable4e = Deliverable4d1[["VendorNumber", "VendorName", "Quantity", "Dollars", "Freight"]]
Deliverable4e.head()

VendorNumber                   VendorName  Quantity  Dollars  Freight
11          1485  CASTLE BRANDS CORP.               320 5,420.41   179.26
16          2242  DELICATO VINEYARDS INC            808 6,646.46   127.05
19          2555  DISARONNO INTERNATIONAL LLC       385 3,506.41   146.86
23          2561  EDRINGTON AMERICAS                136 5,645.24   218.18
28          3924  HEAVEN HILL DISTILLERIES          818 7,079.02   200.02

In [39]:
Deliverable4e["Freight_per_$"] = (Deliverable4e["Freight"] / Deliverable4e["Dollars"] * 100)
Deliverable4e.head()

C:\Users\Tyler Du\AppData\Local\Temp\ipykernel_35464\3108831654.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  Deliverable4e["Freight_per_$"] = (Deliverable4e["Freight"] / Deliverable4e["Dollars"] * 100)


VendorNumber                   VendorName  Quantity  Dollars  Freight  \
11          1485  CASTLE BRANDS CORP.               320 5,420.41   179.26   
16          2242  DELICATO VINEYARDS INC            808 6,646.46   127.05   
19          2555  DISARONNO INTERNATIONAL LLC       385 3,506.41   146.86   
23          2561  EDRINGTON AMERICAS                136 5,645.24   218.18   
28          3924  HEAVEN HILL DISTILLERIES          818 7,079.02   200.02   

    Freight_per_$  
11           3.31  
16           1.91  
19           4.19  
23           3.86  
28           2.83

In [40]:
Deliverable4e["Freight_per_unit"] = (Deliverable4e["Freight"] / Deliverable4e["Quantity"] * 100)
Deliverable4e.head()

C:\Users\Tyler Du\AppData\Local\Temp\ipykernel_35464\1181148165.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  Deliverable4e["Freight_per_unit"] = (Deliverable4e["Freight"] / Deliverable4e["Quantity"] * 100)


VendorNumber                   VendorName  Quantity  Dollars  Freight  \
11          1485  CASTLE BRANDS CORP.               320 5,420.41   179.26   
16          2242  DELICATO VINEYARDS INC            808 6,646.46   127.05   
19          2555  DISARONNO INTERNATIONAL LLC       385 3,506.41   146.86   
23          2561  EDRINGTON AMERICAS                136 5,645.24   218.18   
28          3924  HEAVEN HILL DISTILLERIES          818 7,079.02   200.02   

    Freight_per_$  Freight_per_unit  
11           3.31             56.02  
16           1.91             15.72  
19           4.19             38.15  
23           3.86            160.43  
28           2.83             24.45

In [41]:
# Compared to the findings in 4c, the freight per $ is much higher. This is likely due to the vendors in 4d/4e being limited by the unit sales. This limits the number of sales for the freight and the amount of freight cannot be spread across as much sales. 


# Deliverable 5

In [42]:
Sales = pd.read_csv("C:/Users/Tyler Du/Downloads/SimplifiedData/SimplifiedData/HW4 Sampled Data/SalesFINAL12312016SAMPLE.csv")
Sales.head()

InventoryId  Store  Brand                   Description  \
0   1_HARDERSFIELD_1009      1   1009       Rebel Yell Variety Pack   
1  1_HARDERSFIELD_10238      1  10238   Layer Cake Primitivo Puglia   
2  1_HARDERSFIELD_10239      1  10239       Cannonball Cab Svgn Cal   
3  1_HARDERSFIELD_10266      1  10266  Klinker Brick Old Vine Znfdl   
4   1_HARDERSFIELD_1029      1   1029  Fulton's Harvest Apple Pie L   

         Size  SalesQuantity  SalesDollars  SalesPrice   SalesDate  Volume  \
0  750mL 3 Pk              1         49.99       49.99  2016-01-02  750.00   
1       750mL              2         31.98       15.99  2016-01-02  750.00   
2       750mL              1         13.99       13.99  2016-01-02  750.00   
3       750mL              1         16.99       16.99  2016-01-09  750.00   
4       750mL              1          9.99        9.99  2016-01-03  750.00   

   Classification  ExciseTax  VendorNo                   VendorName  
0               1       0.79      8352  LUXCO INC                    
1               2       0.22      4425        MARTIGNETTI COMPANIES  
2               2       0.11      4425        MARTIGNETTI COMPANIES  
3               2       0.11      9552  M S WALKER INC               
4               1       0.79      3924  HEAVEN HILL DISTILLERIES

In [43]:
Purchases = pd.read_csv("C:/Users/Tyler Du/Downloads/SimplifiedData/SimplifiedData/HW4 Sampled Data/PurchasesFINAL12312016SAMPLE.csv")
Purchases.head()

InventoryId  Store  Brand                   Description   Size  \
0  69_MOUNTMEND_8412     69   8412     Tequila Ocho Plata Fresno  750mL   
1  34_PITMERDEN_5215     34   5215  TGI Fridays Long Island Iced  1.75L   
2  76_DONCASTER_2034     76   2034     Glendalough Double Barrel  750mL   
3      5_SUTTON_3348      5   3348           Bombay Sapphire Gin  1.75L   
4   30_CULCHETH_4903     30   4903          Bacardi Superior Rum  200mL   

   VendorNumber                   VendorName  PONumber      PODate  \
0           105  ALTAMAR BRANDS LLC               8124  2015-12-21   
1          4466  AMERICAN VINTAGE BEVERAGE        8137  2015-12-22   
2           388  ATLANTIC IMPORTING COMPANY       8169  2015-12-24   
3           480  BACARDI USA INC                  8106  2015-12-20   
4           480  BACARDI USA INC                  8106  2015-12-20   

  ReceivingDate InvoiceDate     PayDate  PurchasePrice  Quantity  Dollars  \
0    2016-01-02  2016-01-04  2016-02-16          35.71         6   214.26   
1    2016-01-02  2016-01-07  2016-02-21           9.41         5    47.05   
2    2016-01-02  2016-01-09  2016-02-16          21.32         5   106.60   
3    2016-01-02  2016-01-12  2016-02-05          22.38         6   134.28   
4    2016-01-01  2016-01-12  2016-02-05           2.87        48   137.76   

   Classification  
0               1  
1               1  
2               1  
3               1  
4               1

In [44]:
Deliverable5_1 = pd.merge(left = Sales[['InventoryId', 'Store', 'Brand', 'SalesDollars', 'SalesPrice']], right = Purchases[['InventoryId', 'PurchasePrice', 'Dollars']], how = 'inner', on = ['InventoryId'])
Deliverable5_1.head()

InventoryId  Store  Brand  SalesDollars  SalesPrice  \
0  1_HARDERSFIELD_10238      1  10238         31.98       15.99   
1  1_HARDERSFIELD_10238      1  10238         31.98       15.99   
2  1_HARDERSFIELD_10238      1  10238         31.98       15.99   
3  1_HARDERSFIELD_10238      1  10238         31.98       15.99   
4  1_HARDERSFIELD_10238      1  10238         31.98       15.99   

   PurchasePrice  Dollars  
0          10.66   127.92  
1          10.66   127.92  
2          10.66   127.92  
3          10.66   127.92  
4          10.66    95.94

In [45]:
Deliverable5_1["Profit"] = (Deliverable5_1["SalesPrice"] - Deliverable5_1["PurchasePrice"])
Deliverable5_1 = Deliverable5_1.fillna(0)

Deliverable5_1.head()

InventoryId  Store  Brand  SalesDollars  SalesPrice  \
0  1_HARDERSFIELD_10238      1  10238         31.98       15.99   
1  1_HARDERSFIELD_10238      1  10238         31.98       15.99   
2  1_HARDERSFIELD_10238      1  10238         31.98       15.99   
3  1_HARDERSFIELD_10238      1  10238         31.98       15.99   
4  1_HARDERSFIELD_10238      1  10238         31.98       15.99   

   PurchasePrice  Dollars  Profit  
0          10.66   127.92    5.33  
1          10.66   127.92    5.33  
2          10.66   127.92    5.33  
3          10.66   127.92    5.33  
4          10.66    95.94    5.33

In [46]:
Deliverable5_2 = Deliverable5_1[["Store", "Profit"]].groupby("Store").sum()
Deliverable5_2.sort_values(by='Profit', ascending=False)

Profit
Store             
76    3,436,583.88
73    2,605,387.35
38    2,403,022.20
34    2,303,069.46
66    2,169,189.01
...            ...
29       42,167.37
26       29,307.64
81       24,616.58
46       23,465.70
3        20,825.53

[80 rows x 1 columns]

In [47]:
Deliverable5_3 = Deliverable5_1[["Brand", "Profit"]].groupby("Brand").sum()
Deliverable5_3.sort_values(by='Profit', ascending=False)

Profit
Brand           
1233  522,843.72
3545  445,087.90
3489  390,495.78
3405  364,533.64
2663  353,551.20
...          ...
3375     -435.20
25554    -531.30
33331    -690.78
44714  -1,643.12
25588  -3,236.83

[7401 rows x 1 columns]

This deliverable shows the profit generated for each store and brand for Bibitor. Stores such as 76 and 73 are the best performers, and brands like 1233 and 3535 do the best. We can see this pattern reflected a little bit when we look at beginning and ending inventory with brands such as 1233 and 3545 having high inventories, showing that stores know that these brands are high in demand. The profit numbers give a little more insight on why those inventories might be so high and how it's not necessarily a bad thing to have such high inventory for those products. Overall, it gives a general sense of which stores are doing well and which brands do the best. 

This data is flawed though, if a product wasn't purchased in this year, it wasn't provided in this data set. In addition to that, if a product wasn't sold in this year, it wasn't provided in this data set. I thought that if I included these numbers, it may skew the data if there was products that were sold that were purchased form last year. It would make it seem as if that product was really profitable, but that may not be the case since we didn't purchase any more of it. 
